alphagov/paas-product-pages
# Bump React and Testing Library ecosystems

## What

Bump React and Testing Library ecosystems

Major version bumps of

- react and react-dom to v18
- @testing-library/react to v 13

Removal of `@testing-library/user-event` as it's no longer needed

Updates to analytics test:
- fireEvent instead of simulate clicks as the updated JSDOM throws an error with clicking external links
- tests are now `async`

## How to test

- tests pass
- clone branch
- install updated dependencies (`npm install`)
- build the site (`npm run build`)
- `cd` into `out` folder and serve the page (`npx serve`)
- check page on `localhost:5000` for any issues

There are [expected warnings in CSS compilation](https://github.com/alphagov/govuk-frontend/issues/2238) coming from govuk-frontend
